crates/handler/src/precompile_provider.rs

```diff
@@ -96,7 +96,7 @@ impl<CTX: ContextTr> PrecompileProvider<CTX> for EthPrecompiles {
             }
             Err(e) => {
                 if let PrecompileError::Fatal(_) = e {
-                    return Err(e);
+                    return Err(e.to_string());
```

Comment:
This should be slighly different.

We should catch error on main Err(e) as `Err(PrecompileError::Fatal(e) => return Err(e),` and rest to be matched with `Err(e) =>`

Reply:
Done